.github/workflows/automation-housekeeping.yml

```diff
@@ -38,30 +43,94 @@ jobs:
   branch-protection:
     if: inputs.branch_protection == true
     runs-on: ubuntu-latest
+    concurrency:
+      group: queue
+      cancel-in-progress: false
     steps:
       - name: Checkout Repository
         uses: actions/checkout@master
         with:
           fetch-depth: 0
       - name: Get current branch protection on default branch
-        id: hasBranchProtection
         continue-on-error: true
         run: |
-          gh api repos/{owner}/{repo}/branches/{branch}/protection
-      # Only create branch protection if it doesn't exist, do not modify current settings
+          # Fetch the current branch protection rules
+          gh api repos/{owner}/{repo}/branches/{branch}/protection | \
+            jq '{
+              required_pull_request_reviews: .required_pull_request_reviews | del(.url),
+              enforce_admins: .enforce_admins.enabled,
+              required_status_checks: .required_status_checks | del(.url, .contexts, .contexts_url),
+              restrictions: .restrictions
+            }' > current_protection.json
+
+          # Copy the current protection rules to a working file
+          cat current_protection.json > desired_protection.json
       - name: Enable branch protection on default branch
-        if: ${{ steps.hasBranchProtection.outcome == 'failure' }}
         run: |
-          echo '{
-            "required_pull_request_reviews": {
-              "dismiss_stale_reviews": true,
-              "require_code_owner_reviews": false,
-              "require_last_push_approval": false,
-              "required_approving_review_count": 1
-            },
-            "enforce_admins": true,
-            "required_status_checks": null,
-            "restrictions": null
-          }' > tmp.json
+          # Add the desired PR review rules
+          w=$(jq '.required_pull_request_reviews |= {
+            "dismiss_stale_reviews": true,
+            "require_code_owner_reviews": false,
+            "require_last_push_approval": false,
```

Comment:
Shouldn't require_last_push_approval be true or else you can make changes to a PR after it was approved?

Reply:
Yes you're right :D